packages/synapse-react-client/src/mocks/msw/handlers/tableQueryService.ts

```diff
@@ -0,0 +1,147 @@
+import {
+  Query,
+  QueryBundleRequest,
+  QueryResultBundle,
+} from '@sage-bionetworks/synapse-types'
+import { cloneDeep, omit } from 'lodash-es'
+import {
+  BUNDLE_MASK_LAST_UPDATED_ON,
+  BUNDLE_MASK_QUERY_COLUMN_MODELS,
+  BUNDLE_MASK_QUERY_COUNT,
+  BUNDLE_MASK_QUERY_FACETS,
+  BUNDLE_MASK_QUERY_MAX_ROWS_PER_PAGE,
+  BUNDLE_MASK_QUERY_RESULTS,
+  BUNDLE_MASK_QUERY_SELECT_COLUMNS,
+  BUNDLE_MASK_SUM_FILES_SIZE_BYTES,
+} from '../../../utils/SynapseConstants'
+import BasicMockedCrudService from '../util/BasicMockedCrudService'
+
+const BIT_TO_FIELD_MAP: Record<number, keyof QueryResultBundle> = {
+  [BUNDLE_MASK_QUERY_RESULTS]: 'queryResult',
+  [BUNDLE_MASK_QUERY_COUNT]: 'queryCount',
+  [BUNDLE_MASK_QUERY_SELECT_COLUMNS]: 'selectColumns',
+  [BUNDLE_MASK_QUERY_MAX_ROWS_PER_PAGE]: 'maxRowsPerPage',
+  [BUNDLE_MASK_QUERY_COLUMN_MODELS]: 'columnModels',
+  [BUNDLE_MASK_QUERY_FACETS]: 'facets',
+  [BUNDLE_MASK_SUM_FILES_SIZE_BYTES]: 'sumFileSizes',
+  [BUNDLE_MASK_LAST_UPDATED_ON]: 'lastUpdatedOn',
+}
+
+type QueryAndQueryResultBundle = {
+  id: string
+  query: Omit<Query, 'limit' | 'offset'>
+  result: QueryResultBundle
+}
+const mockTableQueryService = new BasicMockedCrudService<
+  QueryAndQueryResultBundle,
+  'id'
+>({
+  idField: 'id',
+  autoGenerateId: true,
+})
+
+function removeBundleFieldsUsingMask(
+  queryResultBundle: QueryResultBundle,
+  partMask: number,
+): QueryResultBundle {
+  const result = cloneDeep(queryResultBundle)
+  Object.entries(BIT_TO_FIELD_MAP).forEach(([bit, field]) => {
+    if ((partMask & parseInt(bit)) === 0) {
+      delete result[field]
+    }
+  })
+  return result
+}
+
+function applyLimitAndOffset(
+  queryResultBundle: QueryResultBundle,
+  limit: number = 25,
+  offset: number = 0,
+): QueryResultBundle {
+  const result = cloneDeep(queryResultBundle)
+  if (result.queryResult?.queryResults.rows) {
+    result.queryResult.queryResults.rows =
+      result.queryResult.queryResults.rows.slice(offset, offset + limit)
+  }
+  return result
+}
+
+function updateResponseUsingRequest(
+  request: QueryBundleRequest,
+  queryResultBundle: QueryResultBundle,
+): QueryResultBundle {
+  let result = queryResultBundle
+  result = removeBundleFieldsUsingMask(result, request.partMask)
+  result = applyLimitAndOffset(
+    result,
+    request.query.limit,
+    request.query.offset,
+  )
+  return result
+}
```

Comment:
Combine the application of the mask and the application of limit + offset into one method